website/docs/Installation.md

```diff
@@ -1,226 +1,150 @@
-```bash
-pip install "pyautogen[blendsearch]<0.2"
-```
+- Details on how to use the Dockerfile.dev version can be found on the [Contributing](Contribute.md#docker)
 
-Example notebooks:
+## Option 2: Install AutoGen Locally Using Virtual Environment
 
-[Optimize for Code Generation](https://github.com/microsoft/autogen/blob/main/notebook/oai_completion.ipynb)
+### Option a: venv
 
-[Optimize for Math](https://github.com/microsoft/autogen/blob/main/notebook/oai_chatgpt_gpt4.ipynb)
+1. Create and activate a virtual environment:
 
-- #### retrievechat
+   ```bash
+   python3 -m venv pyautogen
+   source pyautogen/bin/activate
+   ```
 
-`pyautogen` supports retrieval-augmented generation tasks such as question answering and code generation with RAG agents. Please install with the [retrievechat] option to use it.
-```bash
-pip install "pyautogen[retrievechat]"
-```
+2. To exit the virtual environment:
 
-RetrieveChat can handle various types of documents. By default, it can process
-plain text and PDF files, including formats such as 'txt', 'json', 'csv', 'tsv',
-'md', 'html', 'htm', 'rtf', 'rst', 'jsonl', 'log', 'xml', 'yaml', 'yml' and 'pdf'.
-If you install [unstructured](https://unstructured-io.github.io/unstructured/installation/full_installation.html)
-(`pip install "unstructured[all-docs]"`), additional document types such as 'docx',
-'doc', 'odt', 'pptx', 'ppt', 'xlsx', 'eml', 'msg', 'epub' will also be supported.
+   ```bash
+   deactivate
+   ```
 
-You can find a list of all supported document types by using `autogen.retrieve_utils.TEXT_FORMATS`.
+### Option b: conda
 
-Example notebooks:
+1. Create and activate a Conda environment:
 
-[Automated Code Generation and Question Answering with Retrieval Augmented Agents](https://github.com/microsoft/autogen/blob/main/notebook/agentchat_RetrieveChat.ipynb)
+   ```bash
+   conda create -n pyautogen python=3.10
+   conda activate pyautogen
+   ```
 
-[Group Chat with Retrieval Augmented Generation (with 5 group member agents and 1 manager agent)](https://github.com/microsoft/autogen/blob/main/notebook/agentchat_groupchat_RAG.ipynb)
+2. To exit Conda environment:
 
-[Automated Code Generation and Question Answering with Qdrant based Retrieval Augmented Agents](https://github.com/microsoft/autogen/blob/main/notebook/agentchat_qdrant_RetrieveChat.ipynb)
+   ```bash
+   conda deactivate
+   ```
 
+### Option c: poetry
 
-- #### Teachability
+1. Initialize and activate a Poetry environment:
 
-To use Teachability, please install AutoGen with the [teachable] option.
-```bash
-pip install "pyautogen[teachable]"
-```
+   ```bash
+   poetry init
+   poetry shell
+   poetry add pyautogen
+   ```
 
-Example notebook:  [Chatting with a teachable agent](https://github.com/microsoft/autogen/blob/main/notebook/agentchat_teachability.ipynb)
+2. To exit the Poetry environment:
 
+   ```bash
+   exit
+   ```
 
+## Python Version Compatibility
 
-- #### Large Multimodal Model (LMM) Agents
```

Comment:
I believe we want to keep the advanced use case examples that require optional dependencies.

Reply:
I agree, that was unintentional on my part. I will add that section and the others that were accidentally removed from the Installation.md file

Reply:
Added the missing sections back. However, should we move the docker information under the Optional Dependencies to the Dockerfile.md?